tobac/utils/general.py

```diff
@@ -259,14 +259,25 @@ def get_spacings(field_in, grid_spacing=None, time_spacing=None):
         # get min and max values of lats and lons
         lat1 = np.min(field_in.coord("latitude").points)
         lat2 = np.max(field_in.coord("latitude").points)
-        lon1 = np.min(field_in.coord("longitude").points)
-        lon2 = np.max(field_in.coord("longitude").points)
-
         # convert decimal degrees to radians
-        lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, lon2, lat2])
-        # haversine formula, note that value is divided by nr of grid cells along lat and lon
-        dlon = (lon2 - lon1) / np.shape(field_in.data)[1]
-        dlat = (lat2 - lat1) / np.shape(field_in.data)[2]
+        lat1, lat2 = map(radians, [lat1, lat2])
+
+        # for 1D lats and lons :
+        lat_axis = -1
+        lon_axis = -1
+        # for 2D lats and lons, check order 
+        if field_in.coord('latitude').points.ndim == 2:
+            if field_in.coords()[1].name() != 'latitude':
```

Comment:
Should we just loop through all of the coordinates and find `latitude` and `longitude` that way?

Reply:
Yes, good suggestion! I changed this part in the code.